logbook-core/src/main/java/org/zalando/logbook/core/DefaultLogbook.java

```diff
@@ -71,18 +80,46 @@ public ResponseWritingStage process(final HttpResponse originalResponse) throws
         }
     }
 
+    private HttpAttributes extractAttributesOrEmpty(final HttpRequest request) {
+        // CompositeAttributeExtractor handles exceptions internally
+        if (attributeExtractor instanceof CompositeAttributeExtractor)
+            return ((CompositeAttributeExtractor) attributeExtractor).extract(request);
```

Comment:
That's a violation of the https://en.wikipedia.org/wiki/Liskov_substitution_principle. Why can't we just use the extractor as-is?

Reply:
Fixed in 4da59d4.